flink-core/src/main/java/org/apache/flink/configuration/StateChangelogOptions.java

```diff
@@ -35,7 +44,7 @@ public class StateChangelogOptions {
                     .withDescription(
                             "Defines the interval in milliseconds to perform "
                                     + "periodic materialization for state backend. "
-                                    + "The periodic materialization will be disabled when the value is negative");
+                                    + "It only takes effect when state.backend.changelog.periodic-materialize.enabled is true");
```

Comment:
`state.backend.changelog.periodic-materialize.enabled` is better to replaced with `PERIODIC_MATERIALIZATION_ENABLED.key()`, if so, the doc can be updated automaticlly if the key is changed in the future.

Reply:
Thanks for the suggestion, I'd like to merge this pr firstly to not block FLINK-33980.
Then I will prepare a quick fix for it, WDYT?

Reply:
Is https://github.com/apache/flink/pull/24025 urgent to be merged? 

- If yes, it's fine.
- If no, it's better to do it in one commit to avoid introuduce unavailable commits in the master branch.

WDYT?

Reply:
Hi @masteryhx, no problem at all. Please feel free to update your PR as needed.

Reply:
Thanks for the confirmation.
I have updated the pr.

Reply:
Thanks for the quick update!